src/test_l1_tvSettings.c

```diff
@@ -364,12 +330,12 @@ void test_l1_tvSettings_positive_RegisterVideoFormatChangeCB (void)
  *
  * **Test Group ID:** Basic : 01@n
  * **Test Case ID:** 06@n
- * 
+ *
```

Comment:
line no 313, As we now using UT_ASSERT_EQUAL, which will proceed even after failures. so checking the allocated data overall would be better and also marking malloc failure in else would help debugging

consider like below snippet. Please check through all functions as we are now using UT_ASSERT_EQUAL

    callbackData.userdata = (char *)malloc(10);
    if (callbackData.userdata) {
        strncpy(callbackData.userdata, "TestData", 10);
        callbackData.cb = tvVideoFormatChangeHandler;
        result = RegisterVideoFormatChangeCB(&callbackData);
        UT_ASSERT_EQUAL(result, tvERROR_NONE);

        free(callbackData.userdata);
        callbackData.userdata = NULL;
    } else {
        UT_LOG("Memory allocation failed for userdata.");
        // Handle the error or assert failure if necessary
    }

Reply:
you can move 
	if(callbackData.userdata){
		free(callbackData.userdata);
		callbackData.userdata = NULL;
	}
this code under 306 condition check

use UT_LOG_ERROR instead of UT_FAIL to record failures

Reply:
Done